ipfs-cluster-service/lock.go

```diff
@@ -0,0 +1,69 @@
+// lock logic heavily inspired by go-ipfs/repo/fsrepo/lock/lock.go
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	filelock "go4.org/lock"
+)
+
+// The name of the file used for locking
+const lockFileName = "cluster.lock"
+
+// Global file, close to relinquish lock access
+var lockCloser io.Closer
+
+func lock(lockPath string) error {
+	// set the lock file within this function
+	logger.Debug("Checking lock")
+	lk, err := filelock.Lock(path.Join(lockPath, lockFileName))
+	if err != nil {
+		lockCloser = nil
+		if err == syscall.EAGAIN {
```

Comment:
I think we are fine just logging the error message whatever it is with logger.Debug() and returning it, without needing to do error parsing.

Right now, the message is shows to the user is:

```
cannot acquire lock: Lock FcntlFlock of /home/hector/go/src/github.com/ipfs/ipfs-cluster/ipfs-cluster-service/d1/cluster.lock failed: resource temporarily unavailable
```

I think regardless of the reason of the failure we can let the user know something like `could not obtain execution lock. If no other cluster process is running, remove ~/path/to/lock, or make sure that folder is writable for the user running cluster. Run with -d for more information about the error`. Perhaps written nicely with line-breaks etc.

Reply:
Sounds good, are you suggesting modifying the error message for greater readability as well, or only adding more logging?

Reply:
I'm suggesting we give an error message that tells clearly the problem and solutions, and we logger.Debug() the original one as it comes from the library.